core/src/main/java/org/mobilitydata/gtfsvalidator/notice/NoticeContainer.java

```diff
@@ -152,4 +173,195 @@ public String exportSystemErrors(boolean isPretty) {
     }
     return noticesByType;
   }
+
+  /**
+   * Exports notices information as a json file.
+   *
+   * @param isPretty will beautify the output if set to true
+   * @param validationNoticeClazz the {@code Class} from which the package name for validation
+   *                             notices will be extracted from
+   * @param validatorClazz the {@code Class} from which the package name for validators will be
+   *                      extracted from
+   * @return the json string file that contains information about all {@code ValidationNotice}s
+   * @throws IOException if the attempt to read class path resources (jar files or directories)
+   * failed.
+   * */
+  public static String exportNoticesSchema(boolean isPretty, Class validationNoticeClazz,
+      Class validatorClazz)
+      throws IOException {
+    Gson gson = isPretty ? PRETTY_GSON : DEFAULT_GSON;
+    JsonObject coreNoticeProperties =
+        extractCoreNoticesProperties(validationNoticeClazz.getPackage().getName());
+    JsonObject mainNoticesProperties =
+        extractMainNoticesProperties(validatorClazz.getPackage().getName());
+    JsonObject root =
+        mergeGsonObjects(ImmutableList.of(coreNoticeProperties, mainNoticesProperties));
+    return gson.toJson(root);
+  }
+
+  /**
+   * Merge multiple {@code com.google.gson.JsonObject} into one.
+   *
+   * @param gsonObjects {@code com.google.gson.JsonObject}s to merge
+   * @return the merged {@code com.google.gson.JsonObject}
+   */
+  private static JsonObject mergeGsonObjects(ImmutableList<JsonObject> gsonObjects) {
+    JsonObject toReturn = new JsonObject();
+    gsonObjects.forEach(gsonObject -> {
+      gsonObject.entrySet().forEach(entry -> {
+        toReturn.add(entry.getKey(), entry.getValue());
+      });
+    });
+    return toReturn;
+  }
+
+  /**
+   * Extract information from notices defined in the core module.
+   *
+   * @param noticePackageName the name of the package that contains the notices
+   * @return the {@code com.google.gson.JsonObject} that contains the {@code ValidationNotice} class
+   * simple name, and the type of each parameter
+   * @throws IOException if the attempt to read class path resources (jar files or directories)
+   * failed.
+   */
+  private static JsonObject extractCoreNoticesProperties(String noticePackageName)
+      throws IOException {
+    Class<?> clazz;
+    JsonObject toReturn = new JsonObject();
+    for (ClassPath.ClassInfo noticeClass :
+        ClassPath.from(ClassLoader.getSystemClassLoader()).getTopLevelClasses(noticePackageName)) {
+      clazz = noticeClass.load();
+      if (ValidationNotice.class.isAssignableFrom(clazz)) {
+        // do not iterate over ValidationNotice.java which is the base class for all validation
+        // notices
+        if (clazz.equals(ValidationNotice.class)){
+          continue;
+        }
+        toReturn.add(
+            noticeClass.getSimpleName(),
```

Comment:
This differs from the error code (e.g., NumberOutOfRangeNotice vs number_out_of_range). Would you like to be consistent? Also below (however, I recommend to avoid code duplication and to add a function that processes a notice class).